src/conn/conn_dhandle.c

```diff
@@ -818,6 +818,15 @@ __wt_conn_dhandle_discard(WT_SESSION_IMPL *session)
         goto restart;
     }
 
+    /*
+     * Return from here if called from recovery flow. Close all dhandles apart from metadata,
```

Comment:
How does this comment fit in? The check below only checks the recovering flag, not anything about history store or metadata.

Reply:
Removed this part of code.